### ISSUE TITLE
Any future plans to build a IOS app?

### ISSUE THREAD

**Issue by ghost** [Ylianst/MeshCentral#4412]
I'm sure other people have asked this question in the past, and I seen a few that the issues were closed a few years back. Just that I would ask if you had any plans on building a IOS app? Would love to be able to use the 2FA like you do on the android side, but IOS works a lot better when you have a mac and ipad, every device works together.

**Comment by si458** [comment]
do you mean an app to control the ios device, or to control other devices from your ios device?
ios has no remote control support but like a view only mode with screen sharing,
and you can already control other devices on ios by opening a browser (like safari) and going to your MC url

**Comment by Ylianst** [comment]
This is not a bad idea. On Android the agent also doubles as a push notification application. It works well, I use it occasionally.

![image](https://user-images.githubusercontent.com/1319013/185689060-3cb09c9f-5408-4ba6-b549-b55c36347135.png)

I don't have the equivalent for iOS and I did start working in Swift a while back to build the MeshCentral Router for macOS. It would be fun to built a MeshCentral app with 2FA for iOS. This said, I am pretty swamped and going on vacation for 6 weeks starting September 7th. If anyone has skills to do this, that would be welcome, but not sure when I will get a chance to work on it.

**Comment by blaine07** [comment]
I do not know how to code; I wouldn’t be of any use to attempt to help but I searched iOS MeshCentral app and it led me here. 

There are a lot us that would appreciate this. :-)

**Comment by DarkxPunk** [comment]
It would be great to see eventually. Frankly the web interface on iOS is largely unusable. Can’t drag windows, scaling is all wrong. I may look into it at some point from a styling perspective to see if I can resolve it, but yeah an app would probably be much easier to fix that stuff.

**Comment by SemoTech** [comment]
Any decision or traction on this?  Would be great to see all MC managed devices on an iOS device, and be able to get statistics and perform remote management tasks on them...  Thanks!

**Comment by dinger1986** [comment]
You can already with a web browser.

I'd be surprised if any large features are introduced now since Intel pulled funding.

**Comment by marclaporte** [comment]
We had a meeting today with a handful of MeshCentral devs. This is a desired feature, and we are looking for someone with experience with iOS apps to coach our developers. Please reach out to me if you can help.

**Comment by SemoTech** [comment]
Thats great news @marclaporte, and while I am not a developer, I do have interface design experience as well as project management and testing experience, so I am happy to provide insights into the design as well as beta test.

**Comment by marclaporte** [comment]
@si458 shared this with us: https://developer.apple.com/documentation/replaykit

**Comment by SemoTech** [comment]
> @si458 shared this with us: https://developer.apple.com/documentation/replaykit

That seems to just be an API for replaying video from a screen in MacOS and iOS...

**Comment by marclaporte** [comment]
We need a developer to step up, or a financial sponsor.

**Comment by silversword411** [comment]
You need to update the title to clarify if this is for agent remote control functionality or administrative controlling functionality.

**Comment by Gamechiefx** [comment]
Can we get an app that remote manages iphones similar to android?

**Comment by marclaporte** [comment]
We need a developer to step up, or a financial sponsor.

**Comment by si458** [comment]
@The-Dev-Ryan has created one here for the community! https://apps.apple.com/us/app/mesh-agent/id6757406184
https://github.com/Ylianst/MeshCentral/pull/7556
its still in beta as the is a few quirks with it which need ironing out

**Comment by SemoTech** [comment]
> [@The-Dev-Ryan](https://github.com/The-Dev-Ryan) has created one here for the community! https://apps.apple.com/us/app/mesh-agent/id6757406184 [#7556](https://github.com/Ylianst/MeshCentral/pull/7556) its still in beta as the is a few quirks with it which need ironing out

Outstanding @si458, thanks for the heads up.  
Looks like the "Mesh Agent" app is designed for iPad not iPhone, which makes sense. I will try it out...

Now if only there was proper Mesh Central MacOS Agent install and operation on MacOS Sequoia & Tahoe, it would be amazing...

**Comment by si458** [comment]
@SemoTech no its designed for iphones as well, i tried on an old iPhone XR (18.7.5) so i know it works!
the just is a few quirks with it which i no @The-Dev-Ryan is going to fix for us, just give him time
(and apple too approve the update etc)

**Comment by SemoTech** [comment]
> [@SemoTech](https://github.com/SemoTech) no its designed for iphones as well, i tried on an old iPhone XR (18.7.5) so i know it works! the just is a few quirks with it which i no [@The-Dev-Ryan](https://github.com/The-Dev-Ryan) is going to fix for us, just give him time (and apple too approve the update etc)

Good to know @si458 and will be patient with it, but the biggest issue now for me is proper support of latest MacOS versions...

**Comment by si458** [comment]
@SemoTech its in the works!
Sadly my mac mini died and can't afford a second hand replacement yet, so need a few months to save up!
Then I can have at it again with the PR the gent made which fixed it in a sense!

**Comment by SemoTech** [comment]
> [@SemoTech](https://github.com/SemoTech) its in the works! Sadly my mac mini died and can't afford a second hand replacement yet, so need a few months to save up! Then I can have at it again with the PR the gent made which fixed it in a sense!

That's great to her @si458.  I look forward to it.
Sorry about your Mac mini, hope you did not loose any important data...

Meantime I've been looking at self-hosted alternatives like Betterdesk (RustDesk) and NetLock-RMM due to the limitations of Mesh Central with newest MacOS.  Sadly they all have their quirks (like needing manual agent config once installed) or need a subscription for signed agent installers which is a big deal on MacOS...

**Comment by Gamechiefx** [comment]
@SemoTech are you saying that this does not work with Tahoe?

**Comment by si458** [comment]
@Gamechiefx yes the new ios app @The-Dev-Ryan made does not work with mac, its ios only

**Comment by SemoTech** [comment]
> [@SemoTech](https://github.com/SemoTech) are you saying that this does not work with Tahoe?

@Gamechiefx The "Mesh Agent" App is for iPad and iPhone.  I already installed it on an iPad. 

What I said does not work with macOS Sequoia and Tahoe is the regular client agent that Mesh Central itself needs installed on a remote Mac to manage it and see the desktop.  Both MacOS Sequoia and Tahoe have tightened security and even the installer of the agent is blocked on those newer OS'.  Even if you bypass MacOS Gatekeeper and get the agent installed it still does not work properly.  We are all waiting on a fix.

Ideally the agent installer and the agent itself will be digitally signed with a valid developer account to ensure no blocking after all the permissions are applied...
